### DISCUSSION TITLE
Would it be possible to edit a laravel config file?

### DISCUSSION THREAD

**Discussion by austinkregel** [nette/php-generator#124] [Q&A] [answered]
:wave: Hi there, I'm really loving this library. Since I found it my mind has been alight with ideas for possible uses.

Most recently, I attempted to use this project to edit a Laravel [`config/app.php`](https://github.com/laravel/laravel/blob/9.x/config/app.php#L157) file. When I had the PhpFile printed out, there were no contents as if the whole file was empty :thinking: 

I thought about it some more and honestly didn't know what I was expecting to have happen. Laravel config files can generally be a variety of things including whole `\Closures`, [method calls](https://github.com/laravel/laravel/blob/9.x/config/view.php#L17), or an [evaluated method chain](https://github.com/laravel/laravel/blob/9.x/config/app.php#L211) which really complicated my thoughts on this matter.

I would really love to be able to edit the config files without having to evaluate the `env` method, any functions, method calls, or chained method calls, however the only way I can find to do so is to evaluate them, turning them into static arrays that lose all the comments.

Since the examples were class like files I wasn't sure if editing a simple php file with a returned array was a supported use case or not. :thinking:

**Comment by dg** [comment, marked as answer]
I think PhpParser would be better suited for this. This library has an API for namespaces, classes, methods, functions, but not for PHP code itself, which it just sees as a strings. With PhpParser you can go further and parse the code of the arrays and modify it.

https://github.com/nikic/PHP-Parser/

**Reply by austinkregel** [reply]
Thank you! I'll take a look at that library
